pkg/controller/oneagent/istio.go

```diff
@@ -84,37 +102,79 @@ func (r *ReconcileOneAgent) reconcileIstioRemoveConfigurations(instance *dynatra
 		seen[istio.BuildNameForEndpoint(instance.Name, ch.Host, ch.Port)] = true
 	}
 
-	vsUpd := r.reconcileIstioRemoveConfiguration(instance, istio.VirtualServiceGVK, listOps, seen, logger)
-	seUpd := r.reconcileIstioRemoveConfiguration(instance, istio.ServiceEntryGVK, listOps, seen, logger)
+	vsUpd := r.reconcileIstioRemoveConfiguration(ic, istio.VirtualServiceGVK, listOps, seen, logger)
+	seUpd := r.reconcileIstioRemoveConfiguration(ic, istio.ServiceEntryGVK, listOps, seen, logger)
 
 	return vsUpd || seUpd
 }
 
-func (r *ReconcileOneAgent) reconcileIstioRemoveConfiguration(instance *dynatracev1alpha1.OneAgent, gvk schema.GroupVersionKind,
-	listOps *client.ListOptions, seen map[string]bool, logger logr.Logger) bool {
+func (r *ReconcileOneAgent) initialiseIstioClient(logger logr.Logger) (*versionedIstioClient.Clientset, error) {
+	ic, err := versionedIstioClient.NewForConfig(r.config)
+	if err != nil {
+		logger.Error(err, fmt.Sprint("istio: failed to initialise client"))
+	}
+	return ic, err
+}
 
-	var list unstructured.UnstructuredList
-	list.SetGroupVersionKind(gvk)
+func (r *ReconcileOneAgent) reconcileIstioRemoveConfiguration(
+	ic *versionedIstioClient.Clientset,
+	gvk schema.GroupVersionKind,
+	listOps *client.ListOptions,
+	seen map[string]bool,
+	logger logr.Logger) bool {
 
-	if err := r.client.List(context.TODO(), listOps, &list); err != nil {
-		logger.Error(err, fmt.Sprintf("istio: failed to list %s objects", gvk.Kind))
+	ic, err := r.initialiseIstioClient(logger)
+	if err != nil {
+		logger.Error(err, fmt.Sprint("istio: failed to initialise client"))
 		return false
 	}
 
-	del := false
-
-	for _, item := range list.Items {
-		if _, inUse := seen[item.GetName()]; !inUse {
-			logger.Info(fmt.Sprintf("istio: removing %s: %v", gvk.Kind, item.GetName()))
-			if err := r.client.Delete(context.TODO(), &item); err != nil {
-				logger.Error(err, fmt.Sprintf("istio: failed to delete %s", gvk.Kind))
-				continue
+	if gvk == istio.ServiceEntryGVK {
```

Comment:
I'd suggest splitting the function by GVK since no logic is shared between the two if branches.